apps/meteor/server/services/federation/infrastructure/rocket-chat/hooks/index.ts

```diff
@@ -190,9 +185,16 @@ export class FederationHooks {
 	}
 
 	public static async afterRoomRoleChanged(federationRoomService: FederationRoomServiceSender, data?: Record<string, any>) {
-		if (!data || !settings.get('Federation_Matrix_enabled')) {
+		if (!data) {
+			return;
+		}
+
+		if (!isFederationEnabled()) {
 			return;
 		}
+
+		throwIfFederationNotReady();
```

Comment:
Are we checking and throwing here only if it's not ready yet because we were just (early) returning if federation is not enabled, right?

Reply:
Yes